### DISCUSSION TITLE
application permissions

### DISCUSSION THREAD

**Discussion by VilmanVladimir** [MonoGame/MonoGame#8205] [Android]
Hi all
Help with the problem that has arisen. Previously, to enable write permission to local storage, I wrote this:

…
using Android.Support.V4.Content;
using Android.Support.V4.App;
…
…
protected override void Initialize()
{
//Request to enable permissions
if (ContextCompat.CheckSelfPermission(Activity, Manifest.Permission.ReadExternalStorage) != (int)Permission.Granted)
{
// Permission has never been accepted
// So, I ask the user for permission
ActivityCompat.RequestPermissions(Activity, new String { Manifest.Permission.ReadExternalStorage }, PERMISSION_REQUEST_READ_EXTERNAL_STORAGE);
}
base.Initialize();
}

And everything worked well.

And after switching to version 3.8.1 errors appeared
![9dba5b6ceeaf63868631c4c0df8f5bac9eaae74a](https://github.com/MonoGame/MonoGame/assets/160932805/df484def-6bc9-4daa-b732-9d4d4f723229)

And in the method:
![7be50d5c56ddc124217ba916db3c5f66e9688396](https://github.com/MonoGame/MonoGame/assets/160932805/8b4a4b89-cd91-4010-aa65-85f3248d6e3a)

Has anyone encountered this? And how did you solve the problem?

**Comment by dellis1972** [comment]
You are using the old Android Support librarys you will need to upgrade to use the AndroidX based libraries to get this to work on newer API levels. The old Support stuff has been deprecated for quite a while now.
